### DISCUSSION TITLE
Update the session exclusively from the server

### DISCUSSION THREAD

**Discussion by musjj** [nextauthjs/next-auth#11294] [Ideas]
### Goals

Update the session data from the server, while preventing the client from doing the same.

### Background

Server-side session updates are currently possible using a combination of `jwt` callbacks:
https://github.com/nextauthjs/next-auth/blob/5c49f958e0d6b5e78c574298fd9335ee04d4e90b/apps/dev/nextjs/auth.config.ts#L48-L51

And `unstable_update`:
https://github.com/nextauthjs/next-auth/blob/5c49f958e0d6b5e78c574298fd9335ee04d4e90b/apps/dev/nextjs/app/page.tsx#L1
https://github.com/nextauthjs/next-auth/blob/5c49f958e0d6b5e78c574298fd9335ee04d4e90b/apps/dev/nextjs/app/page.tsx#L22-L29

It works, but it really doesn't feel right. With the current setup, the client can inject any data they want to the current session token, due to the `jwt` callback. But this setup will not work without the having that callback.

### Proposal

Have a server-exclusive method for updating the session data. 

The `jwt` callback method above should be optional for users who *really* want to have a client-side method for updating the session.

I'm hoping that the current state is temporary (they are marked as `unstable_*` after all) and that they will be improved in the future.

**Comment by corned-beefhash** [comment]
when i try this in nextjs, i get errors about how "Cookies can only be modified in a Server Action"  even though it is in a server action.  
also, how can a client inject data into the session?

**Reply by musjj** [reply]
Use the `update` function returned by `useSession()`.

**Reply by corned-beefhash** [reply]
Ahh, I do not use useSession anywhere at the moment.  Hopefully they make them distinct, but in mean time perhaps you can use a key of some kind in your unstable_update,   and then in your jwt,  return token early if key is wrong or missing?  No idea, I am pretty new to all of this.
